### ISSUE TITLE
AuthData Migration

### ISSUE THREAD

**Issue by felix-dumit** [ParsePlatform/parse-server#3360]
I'm migrating a few of my users from the old Parse.com data into my new `parse-server`. 
I don't want to migrate all the data via mongodb because I need to perform a custom data migration since I updated my data model.

However, I'm having trouble importing the authData in two different scenarios:

### 1) Expired Facebook AuthData
If a user's Facebook `access_token` has expired I get the error `Facebook Auth is invalid for this user`

### 2) Digits User
I did not use Parse's 3rd party authentication, and instead implemented login with digits using a custom cloud function, so I only stored the user's `id` and no `auth_token`. And I cannot save the `authData` with just: `{"twitter": {"id": "ID_HERE"}}`

---
In both of these cases, how can I create a new user, on my new parse-server instance so that if the original user (facebook or digits) logs in, he will be logged into this user as opposed to having a new user created?
Basically, I don't really care about persisting the access tokens across the migration, I just want to persist the link from the account (facebook or digits) to the user instance.

Thank you.

**Comment by flovilmart** [comment]
You should not need to create new users, because the `id` from the 3rd party service is there, any user that will login with Facebook or twitter should properly be mapped to the original user.

**Comment by felix-dumit** [comment]
Hi @flovilmart , I think maybe I wasn't clear enough in my question.
The existing users are on one server, and I want to move them to my hosted `parse-server`. 

I cannot save these users on the new server with an invalid authData (just using the `id`).

**Comment by flovilmart** [comment]
> I cannot save these users on the new server with an invalid authData

What do you mean you cannot? The migration fails?

**Comment by felix-dumit** [comment]
I don't want to do a full data migration, so I'm performing this migration "manually", by querying from the old parse server, modifying the data, and saving to the new parse server.

I mean I cannot save because If I try to save an user with the authData as below:
```
{authData: {"twitter": { "id": "SOME_ID"}} 
```
It will fail saving with `Twitter auth is invalid for this user`. 
(same thing for a facebook user with expired access token or just with the id)

**Comment by flovilmart** [comment]
You should do that migration at the mongoDB level, not at the Parse format level.

**Comment by felix-dumit** [comment]
So I tried messing with the "_auth_data_facebook" and "_auth_data_twitter" fields directly in mongoDB, and it will work in that it will populate the authData correctly and it will even link to the account when I log in. 
The only downside is that I cannot save this user again with parse (will fail authData validation) until the user has successfully re-logged in.
That's not a huge issue for me, but do you have any other ideas?

**Comment by flovilmart** [comment]
What I mean is that you should probably migrate the mongoDB to another mongoDB, without thinking about Parse, and then delete all entries that don't match what you want...

**Comment by felix-dumit** [comment]
You mean then after I'm done "migrating" in this new mongoDB instance, I would then use that in my parse-server?
I suppose that could work, I would still end up doing having to do what I mentioned in my last comment to change the authData.

If that's pretty much the only way to do it feel free to close this then.

**Comment by flovilmart** [comment]
yeah, just make a copy of the mongoDB with [mongodump/mongorestore](https://docs.mongodb.com/manual/core/backups/) or [copyDB] (http://stackoverflow.com/questions/5495137/how-do-i-copy-a-database-from-one-mongodb-server-to-another#5495401)

On your duplicated DB, you can pretty much do whatever you like, as it will be a clean copy. IF anything goes wrong, you still will have the dumps and be able to restore.
